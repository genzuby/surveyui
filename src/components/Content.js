import React from "react";
// for Adapted web design
import { isMobile } from "react-device-detect";
import { CONTENTTEXT } from "../styles/BodyStyles";

const styles = {
  fontSize: "calc(1.2rem + .2vw)"
};
// render body or question property
const Content = ({ contentVal }) => {
  const adaptedText = () => {
    if (isMobile) {
      return <CONTENTTEXT>{contentVal}</CONTENTTEXT>;
    } else {
      return <CONTENTTEXT style={styles}>{contentVal}</CONTENTTEXT>;
    }
  };

  return <div className="Content">{adaptedText()}</div>;
};

export default Content;
