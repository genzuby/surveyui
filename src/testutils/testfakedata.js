export const surveylist = [
  {
    questionID: "page-welcome",
    type: "Page",
    heading: "Welcome to the survey",
    body: "You can start the survey by clicking the next button",
    next: "question-email"
  },
  {
    questionID: "page-uneligible",
    type: "Page",
    heading: "Not eligible",
    body: "We're sorry, but you are not eligible to use our producs/services"
  },
  {
    questionID: "page-finish",
    type: "Page",
    heading: "Thank you",
    body: "You have completed the survey"
  },
  {
    questionID: "question-email",
    type: "Text",
    question: "What is your email address?",
    next: "question-age"
  },
  {
    questionID: "question-age",
    type: "Radio",
    question: "Are you 18 or older?",
    choices: [
      {
        label: "Yes",
        value: "yes"
      },
      {
        label: "No",
        value: "no",
        next: "page-uneligible"
      }
    ],
    next: "question-problems"
  },
  {
    questionID: "question-problems",
    type: "Checkbox",
    question: "Do you suffer from any of the following problems?",
    choices: [
      {
        label: "I have dental implants",
        value: "dental_implants"
      },
      {
        label: "I have missing teeth",
        value: "missing_teeth"
      },
      {
        label: "I have unfilled cavities",
        value: "unfilled_cavities"
      },
      {
        label: "I have swollen or inflamed gums",
        value: "swollen_or_inflamed_gums"
      },
      {
        label: "I have retainers",
        value: "retainers"
      },
      {
        label: "None of the above",
        value: "no"
      }
    ],
    next: "page-finish"
  }
];

export const initData = [
  {
    surveyID: "345f52c7-78f6-4cc4-9f8a-0490744f7da3",
    surveyItems: [
      {
        questionID: "page-welcome",
        type: "Page",
        heading: "Welcome to the survey",
        body: "You can start the survey by clicking the next button",
        next: "question-email"
      },
      {
        questionID: "page-uneligible",
        type: "Page",
        heading: "Not eligible",
        body:
          "We're sorry, but you are not eligible to use our producs/services"
      },
      {
        questionID: "page-finish",
        type: "Page",
        heading: "Thank you",
        body: "You have completed the survey"
      },
      {
        questionID: "question-email",
        type: "Text",
        question: "What is your email address?",
        next: "question-age"
      },
      {
        questionID: "question-age",
        type: "Radio",
        question: "Are you 18 or older?",
        choices: [
          {
            label: "Yes",
            value: "yes"
          },
          {
            label: "No",
            value: "no",
            next: "page-uneligible"
          }
        ],
        next: "question-problems"
      },
      {
        questionID: "question-problems",
        type: "Checkbox",
        question: "Do you suffer from any of the following problems?",
        choices: [
          {
            label: "I have dental implants",
            value: "dental_implants"
          },
          {
            label: "I have missing teeth",
            value: "missing_teeth"
          },
          {
            label: "I have unfilled cavities",
            value: "unfilled_cavities"
          },
          {
            label: "I have swollen or inflamed gums",
            value: "swollen_or_inflamed_gums"
          },
          {
            label: "I have retainers",
            value: "retainers"
          },
          {
            label: "None of the above",
            value: "no"
          }
        ],
        next: "page-finish"
      }
    ]
  }
];

export const testAnswer = {
  userId: "test3333@gmail.com",
  userInfo: {
    under18: false,
    finished: true
  },
  answeredlist: {
    surveyID: "345f52c7-78f6-4cc4-9f8a-0490744f7da3",
    surveyItems: [
      {
        questionID: "question-problems",
        type: "Checkbox",
        choices: ["dental_implants", "unfilled_cavities"]
      }
    ]
  }
};
