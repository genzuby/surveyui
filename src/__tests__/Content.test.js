import React from "react";
import { mount } from "enzyme";
import "../testutils/setup";
import Content from "../components/Content";

describe("Render Content", () => {
  let wrapper;

  beforeEach(() => {
    const props = {
      contentVal: "Test Content Page"
    };
    wrapper = mount(<Content {...props} />);
  });

  describe("Should render without error", () => {
    it("renders correctly", () => {
      expect(wrapper).toMatchSnapshot();
    });

    it("should render one <p>", () => {
      expect(wrapper.find("p")).toHaveLength(1);
    });

    it("Should render with proper text", () => {
      wrapper.setProps({ contentVal: "Test Content Page" });
      expect(wrapper.text()).toEqual("Test Content Page");
    });
  });
});
