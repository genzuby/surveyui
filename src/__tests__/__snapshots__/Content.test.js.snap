// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Buttons Should render without error renders correctly 1`] = `undefined`;

exports[`Render Content Should render without error renders correctly 1`] = `ReactWrapper {}`;
